### DISCUSSION TITLE
filter -> equal and not equal -> empty string or null -> working in inverse way

### DISCUSSION THREAD

**Discussion by rafakaiber** [primefaces/community#1019] [PrimeNG]
I'm working with primeNG (17.0.0) table in my angular TS (17.0.0).
All filters are working as expected, but on for equal and not equal for empty string is working in inverse way.
If I filter or clear the filter at this state: 
1-  not equals, empty string or null, it returns no value and should return all values that has any value.
2- equals, empty string or null, it returns all values and shoul return fields that has no value.

Is some one seeing same problem?

**Comment by cetincakiroglu** [comment]
Hi,

Could you please share a stackblitz reproducer so we can identify the issue clearly?

**Reply by rafakaiber** [reply]
please check my message below with reference!

**Comment by rafakaiber** [comment]
Go to https://primeng.org/table#filter-menu
There select filter in Country column, change the filter to Not Equal and click Apply button.
you will see all data is gone, and if you change to Equals, with empty string and apply all data will return.

attaching a Screen shot for reference:

![image](https://github.com/primefaces/community/assets/118986864/18acddfd-47d2-4c89-b952-e8a14cf43fc0)

after Apply click you see no values:

![image](https://github.com/primefaces/community/assets/118986864/80f51659-544a-45cf-8f88-5c63698bf6b6)


After that yo can try the inverse, change the filter to Equal and Apply, all data returned.
I believe it will make it clear to see the issue!

If need more info please let me know!

**Reply by cetincakiroglu** [reply]
Hi,

Thanks for the detailed explanation, could you please open an issue in GitHub repo and let me know so we can add it to the milestone and address the issue in upcoming releases?
